Bottom: 03abf75c7ba913f7df51535ce8f5e1a7260f255b
Top:    504e16c4a943018e7db2bae25fa776c9f2624078
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 19:58:56 +0000

polygon1.c: Temporarily disable the seg_in_region test

The code does not handle the bounding region of arcs yet


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 76d8657..7847be6 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -637,6 +637,7 @@ adjust_tree (rtree_t * tree, struct seg *s)
   return 0;
 }
 
+#if 0  /* TURNS OUT DISABLING THIS ACTUALLY (VERY MARGINALLY) SPEEDS THINGS UP! */
 /*
  * seg_in_region()
  * (C) 2006, harry eaton
@@ -658,6 +659,7 @@ seg_in_region (const BoxType * b, void *cl)
     return 0;
   return 1;			/* might intersect */
 }
+#endif
 
 /* Prepend a deferred node-insersion task to a list */
 static insert_node_task *
@@ -881,7 +883,7 @@ contour_bounds_touch (const BoxType * b, void *cl)
       info.tree = rtree_over->tree;
       if (info.tree)
 	if (UNLIKELY (r_search (info.tree, &info.s->box,
-				seg_in_region, seg_in_seg, &info)))
+				NULL /* seg_in_region */, seg_in_seg, &info)))
 	  assert (0); /* XXX: Memory allocation failure */
     }
   while ((av = NEXT_EDGE (av)) != &looping_over->head);
